LibreNMS/Interfaces/Discovery/VlanDiscovery.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace LibreNMS\Interfaces\Discovery;
+
+interface VlanDiscovery
+{
+    /**
+     * Discover processors.
+     * Returns an array of LibreNMS\App\Vlan && LibreNMS\App\PortVlan objects
```

Comment:
Why are there two types of data mixed together?  Perhaps there should be two discovery methods?
On newer modules, I've been returning collections of Models to help keep the data more consistent.

Reply:
Something like this, what do you think?

discoverVlans(): Collection<\App\Models\Vlan>;
discoverVlanPorts(): Collection<\App\Models\PortVlan>;

Reply:
> discoverVlans(): Collection<\App\Models\Vlan>;
> discoverVlanPorts(Collection $vlans): Collection<\App\Models\PortVlan>;

hi @murrant 
correct me if i am wrong ...
my intention was to put whole vlan discovery in one function
since it have 
[basic] data, name,type
and
[ports] data, tagged,portid,etc
i needed to return somehow this dual data, and array was handy
one function, one array,then split & process at centralized place

but, if you insist to split them in two function, discoverBasicVlan, discoverPortVlan ....

how to be then ?

Reply:
I'm just trying to make things match our other code so they are more consistent. No discovery interfaces we currently have return multiple sets of data.

The data seems unconnected to me so it seems separate functions makes sense.  You've gone through all the code though, so you might have more context to me.

If we do want to return both, I don't like using arrays for that I would much prefer a data transfer object to hold the data in a structured way that we can expect.  But again this would be unique, different from every other module we currently have.